app/validators/questions/calculated_question_validator.py

```diff
@@ -19,3 +22,31 @@ def validate_calculations(self):
             for answer_id in calculation["answers_to_calculate"]:
                 if answer_id not in answer_ids:
                     self.add_error(self.ANSWER_NOT_IN_QUESTION, answer_id=answer_id)
+
+    def _validate_answer_is_numeric(self, answer_id):
+        answer_type = self.schema.get_answer_type(answer_id)
+
+        if ANSWER_TYPE_TO_JSON_TYPE[answer_type.value] != TYPE_NUMBER:
+            self.add_error(
+                self.ANSWER_TYPE_FOR_CALCULATION_TYPE_INVALID.format(
+                    answer_type=ANSWER_TYPE_TO_JSON_TYPE[answer_type.value],
+                ),
+                identifier=answer_id,
+            )
+
+    def validate_calculations_value_source_is_numeric(self):
+        """
+        Validates that source answer is of number type
+        """
+        for calculation in self.question.get("calculations"):
+            value = calculation.get("value")
+
+            if answer_id := calculation.get("answer_id"):
+                self._validate_answer_is_numeric(answer_id)
+
+            elif value and value.get("source"):
+                answer_id = value.get("identifier")
+                # Calculated summary value source is validated elsewhere and must be of a number type
+
+                if value.get("source") == "answers":
+                    self._validate_answer_is_numeric(answer_id)
```

Comment:
Only a suggestion as it may help author. Might be worth passing in the question Id so you can log it out in the error so author can see what answer_id isn't numeric and which question it is referenced in. See what @berroar thinks.

Reply:
Added question id, thanks.